feat: add AnimatedSideMenu component with demo and docs

srikarnarayanempati opened the pull request:
# Description & Technical Solution

This PR adds a new AnimatedSideMenu component to the UI library.
The goal was to provide a sleek, theme-aware sidebar menu with smooth open/close transitions and glowing toggle button for a modern UX.

Motivation & Context

- Existing components didn’t include a menu system that combined motion, theme-awareness, and interactive glow effects.

- This addition enhances the library by providing a ready-to-use navigation pattern often needed in portfolios, dashboards, and apps.

Technical Solution

- Implemented with React + Framer Motion for smooth transitions (AnimatePresence + motion).

- Button state toggles between MENU and CLOSE with animated text transitions.

- Theme-aware styling:

- Light mode → button starts dark, expands to light.

- Dark mode → button starts light, expands to dark.

- Added soft glow + shadow effects for both button and panel, adjustable via box-shadow.

- Expansion panel animates from the top-right corner with scaling transitions, showing menu items (HOME, ABOUT,                PROJECTS, etc.).

A demo file and docs (MDX) were created for usage examples and installation guidance.

# Checklist

 -[✅]Code is self-contained and readable.

 -[✅]Theme-aware styles tested in both light and dark modes.

 -[✅]Demo and docs (animatedsidemenu-demo.tsx + .mdx) added.

 -[✅]Verified build and preview work locally.

 -[ ]Already rebased against main branch.

# Screenshots
<img width="708" height="878" alt="Screenshot 2025-08-17 at 3 59 44 AM" src="https://github.com/user-attachments/assets/4ad4f605-825b-4a28-97bf-ef2351d20dd2" />
<img width="708" height="878" alt="Screenshot 2025-08-17 at 3 59 59 AM" src="https://github.com/user-attachments/assets/0618fceb-a30f-42ef-9d1d-916e3914f68c" />

Adityakishore0 commented:
Hey @srikarnarayanempati, I went through the MDX docs and noticed a few issues:

### 1. Missing dependencies
- The docs don’t mention that `framer-motion` is required.
- Anyone copying the code won’t know they need to install it.  
**Suggestion:** Add `framer-motion` to the dependencies section.

### 2. Hardcoded toggle button
- The button currently shows `"MENU"` / `"CLOSE"` directly.
- Problem: Users cannot customize the button text, icons, or fully replace it.  
**Suggestion:** Use a `toggleChildren` prop so the toggle content can be anything.

### 3. Hardcoded panel content
- Links like `HOME`, `ABOUT`, `PROJECTS`, etc. are fixed.
- Problem: Users cannot add their own items, links, or components.  
**Suggestion:** Use a `panelChildren` prop to allow fully custom content. Users will handle scroll, layout, and styling inside the panel themselves.

### 4. Fixed panel size
- Panel width and height are hardcoded (`w-[400px] h-[350px]`).
- Problem: Users cannot adjust size or styling.  
**Suggestion:** Let users pass `className` for the panel and `buttonClassName` for the toggle button.

### 5. Manual dark mode handling
- Uses `document.documentElement.classList.contains("dark")` and inline styles.
- Problem: Button component  already handle colors.  
**Suggestion:** Remove manual dark mode logic and rely on Tailwind.

### 6. Not fully generic / TypeScript safe
- Component assumes defaults for everything and lacks a typed interface.  
**Suggestion:** Add TypeScript props interface with `toggleChildren`, `panelChildren`, `className`, and `buttonClassName`.

**Goal:** Make this component fully **reusable, customizable, and flexible** while keeping it TypeScript-safe.

srikarnarayanempati commented:
Hi @Adityakishore0,

I’m working on resolving the issues you mentioned, but I’ll need a little more time to complete them. Also, regarding the component name change to ToggleVault, I feel that AnimatedSideMenu is quite descriptive and suitable for its functionality. Would it be possible to retain the current name instead of changing it?

Thank you for your understanding.

srikarnarayanempati commented:
Hey @Adityakishore0 


 **Updates made based on review feedback:**

- Renamed files to follow repo conventions (toggle-vault.tsx, togglevault-demo.tsx, etc).
- Updated navitems.ts and registry.json references.
- Fixed type-safety issues in ToggleVaultTrigger and ToggleVaultContent.
- Made dark/light mode fully customizable via props (no hardcoding).
- Added close on link click functionality for better UX.
- Improved layout so the toggle button + expansion panel render correctly inside the preview area.
- Added soft glow effects matching respective theme colors.
- Updated MDX docs (usage + props table) to reflect new API.

As you told before ,if your implementation ensures stronger type safety and full customization, I’m completely fine with moving forward with yours. Otherwise, feel free to adjust my current changes so we can finalize and get this merged.

srikarnarayanempati commented:
Thanks a lot for the detailed review and improvements .
I really appreciate you taking the time to point out the gaps and make the component fully customizable and accessible. It’s great to learn where I can refine things further, especially around accessibility, flexibility, and render-props usage.

This was a really good learning experience for me — I’ll keep these points in mind for future contributions. Excited to contribute more!
